src/inc/eventtracebase.h

```diff
@@ -77,33 +83,35 @@ enum EtwThreadFlags
 // if the fields in the event are not cheap to calculate
 //
 #define ETW_EVENT_ENABLED(Context, EventDescriptor) \
-    ((MCGEN_ENABLE_CHECK(Context, EventDescriptor)) || EVENT_PIPE_ENABLED())
+    ((MCGEN_ENABLE_CHECK(Context, EventDescriptor)) || EventEnabled##EventDescriptor())
 
 //
 // Use this macro to check if a category of events is enabled
 //
 
 #define ETW_CATEGORY_ENABLED(Context, Level, Keyword) \
-    ((Context.IsEnabled && McGenEventProviderEnabled(&Context, Level, Keyword)) || EVENT_PIPE_ENABLED())
+    ((Context.IsEnabled && McGenEventProviderEnabled(&Context, Level, Keyword)) || EventPipeHelper::Enabled(EVENT_PROVIDER_IDX_##Context, Level, Keyword))
```

Comment:
This probably won't perform as well as we need it to. When tracing configuration is disabled the runtime still has to call this code in order to know that it is disabled, thus we pay for this check frequently in performance sensitive paths. In an ideal world this check would be just a few cpu cycles. We can probably do worse than that (10s, maybe even low 100s?) without it showing up as an obvious performance regression, but beyond that I'd be quite surprised if you could avoid seeing bad perf results.

Right now I think this code is doing an int -> string conversion, an EventPipe::GetLock() acquisition, a linked list search, and multiple string comparisons. ETW achieved high speed by [registering known providers in advance](https://github.com/dotnet/coreclr/blob/master/src/vm/eventtrace.cpp#L4148) and [saving direct references into globals](https://github.com/dotnet/coreclr/blob/master/src/vm/eventtrace.cpp#L4307). I would try to follow that plan.